src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java

```diff
@@ -35,52 +40,93 @@
-                        .payload(modifiedRequest)
+                        .body(mapper.encodeToBytes(modifiedBidRequest))
+                        .payload(modifiedBidRequest)
                         .build());
     }
 
-    private static BidRequest modifyRequest(BidRequest bidRequest, String masterTagId) {
+    private ExtImpAdview parseExtImp(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("invalid imp.ext");
+        }
+    }
+
+    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
+        return shouldConvertBidFloor(imp.getBidfloor(), imp.getBidfloorcur())
+                ? convertBidFloorCurrency(imp.getBidfloor(), request, imp.getId(), imp.getBidfloorcur())
+                : imp.getBidfloor();
+    }
+
+    private static boolean shouldConvertBidFloor(BigDecimal bidFloor, String bidFloorCur) {
+        return BidderUtil.isValidPrice(bidFloor) && !StringUtils.equalsIgnoreCase(bidFloorCur, BIDDER_CURRENCY);
+    }
+
+    private BigDecimal convertBidFloorCurrency(BigDecimal bidFloor,
+                                               BidRequest bidRequest,
+                                               String impId,
+                                               String bidFloorCur) {
+        try {
+            return currencyConversionService
+                    .convertCurrency(bidFloor, bidRequest, bidFloorCur, BIDDER_CURRENCY);
+        } catch (PreBidException e) {
+            throw new PreBidException(String.format(
+                    "Unable to convert provided bid floor currency from %s to %s for imp `%s`",
+                    bidFloorCur, BIDDER_CURRENCY, impId));
+        }
+    }
+
+    private static BidRequest modifyRequest(BidRequest bidRequest, String masterTagId, BigDecimal resolvedBidFloor) {
         return bidRequest.toBuilder()
-                .imp(modifyImps(bidRequest.getImp(), masterTagId))
+                .imp(modifyImps(bidRequest.getImp(), masterTagId, resolvedBidFloor))
+                .cur(Collections.singletonList("USD"))
                 .build();
     }
 
-    private static List<Imp> modifyImps(List<Imp> imps, String masterTagId) {
+    private static List<Imp> modifyImps(List<Imp> imps, String masterTagId, BigDecimal resolvedBidFloor) {
         final List<Imp> modifiedImps = new ArrayList<>(imps);
-        modifiedImps.set(0, modifyImp(imps.get(0), masterTagId));
+        modifiedImps.set(0, modifyImp(imps.get(0), masterTagId, resolvedBidFloor));
         return modifiedImps;
     }
 
-    private static Imp modifyImp(Imp imp, String masterTagId) {
+    private static Imp modifyImp(Imp imp, String masterTagId, BigDecimal resolvedBidFloor) {
         return imp.toBuilder()
                 .tagid(masterTagId)
+                .bidfloor(resolvedBidFloor)
+                .bidfloorcur(BIDDER_CURRENCY)
```

Comment:
`    private BigDecimal resolveBidFloor(BidRequest request, Imp imp) {
        final BigDecimal bidFloor = imp.getBidfloor();
        final String bidFloorCur = imp.getBidfloorcur();

        return shouldConvertBidFloor(bidFloor, bidFloorCur)
                ? convertBidFloorCurrency(bidFloor, bidFloorCur, imp.getId(), request)
                : bidFloor;
    }`
 According to this, we are only converting floor if it is valid, if now, we are returning floor, we got in request.
 If we are getting null in bidfloor field, we are passing it without a pain :)
 
 In PBS-Go its done same way, but they don't touch it at all, if it is not valid.

Reply:
That's fine about `bidfloor == null` logic here. I'm worrying about `bidfloorcur` which will be set unconditionally as `BIDDER_CURRENCY`.
In case of PBS-Go, guys set it only if `bidfloor` is valid and converted to new value.